test/integration/features/utilities/utils.py

```diff
@@ -147,3 +147,10 @@ def search_dos_db(query: str) -> list:
     rows = db_cursor.fetchall()
     db_cursor.close()
     return rows
+
+
+def generate_correlation_id(context: dict) -> str:
```

Comment:
could you add extra param here called suffix, which you could pass in your case as "BAD REQUEST" should still trigger but also mean correlation id is still tied to the run and the scenario ?